src/vulkan-renderer/world/cube.cpp

```diff
@@ -395,23 +408,22 @@ std::vector<PolygonCache> Cube::polygons(const bool update_invalid) const {
     std::vector<PolygonCache> polygons;
     polygons.reserve(count_geometry_cubes());
 
-    std::function<void(std::shared_ptr<const world::Cube>)> collect;
     // post-order traversal
-    collect = [&collect, &polygons, &update_invalid](std::shared_ptr<const world::Cube> cube) {
-        if (cube->type() == world::Cube::Type::OCTANT) {
-            for (const auto &child : cube->childs()) {
-                collect(child);
+    std::function<void(const Cube &)> collect = [&collect, &polygons, &update_invalid](const Cube &cube) {
```

Comment:
```suggestion
    static std::function<void(const Cube &)> collect = [&collect, &polygons, &update_invalid](const Cube &cube) {
```

Reply:
This can't be static

```cpp
/home/johannes/Code/inexor/warning-cleanup/vulkan-renderer/src/vulkan-renderer/world/cube.cpp:412:58: warning: capture of variable ‘collect’ with non-automatic storage duration
  412 |     static std::function<void(const Cube &)> collect = [&collect, &polygons, &update_invalid](const Cube &cube) {
      |                                                          ^~~~~~~
/home/johannes/Code/inexor/warning-cleanup/vulkan-renderer/src/vulkan-renderer/world/cube.cpp:412:46: note: ‘std::function<void(const inexor::vulkan_renderer::world::Cube&)> collect’ declared here
  412 |     static std::function<void(const Cube &)> collect = [&collect, &polygons, &update_invalid](const Cube &cube) {
      |                                              ^~~~~~~
```